src/meta-srv/src/event_recorder/region_migration_event.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::fmt::Display;
+
+use api::v1::value::ValueData;
+use api::v1::{
+    ColumnDataType, ColumnSchema, Row, RowInsertRequest, RowInsertRequests, Rows, SemanticType,
+};
+use chrono::Utc;
+use common_procedure::error::Error;
+use common_procedure::ProcedureId;
+
+use crate::event_recorder::Event;
+use crate::procedure::region_migration::manager::RegionMigrationProcedureTask;
+
+pub const REGION_MIGRATION_EVENTS_TABLE_NAME: &str = "region_migration_events";
+
+pub const REGION_MIGRATION_EVENTS_TABLE_PROCEDURE_ID_COLUMN_NAME: &str = "procedure_id";
+pub const REGION_MIGRATION_EVENTS_TABLE_REGION_ID_COLUMN_NAME: &str = "region_id";
+pub const REGION_MIGRATION_EVENTS_TABLE_FROM_DATANODE_ID_COLUMN_NAME: &str = "from_datanode_id";
+pub const REGION_MIGRATION_EVENTS_TABLE_FROM_DATANODE_ADDR_COLUMN_NAME: &str = "from_datanode_addr";
+pub const REGION_MIGRATION_EVENTS_TABLE_TO_DATANODE_ID_COLUMN_NAME: &str = "to_datanode_id";
+pub const REGION_MIGRATION_EVENTS_TABLE_TO_DATANODE_ADDR_COLUMN_NAME: &str = "to_datanode_addr";
+pub const REGION_MIGRATION_EVENTS_TABLE_TRIGGER_REASON_COLUMN_NAME: &str = "trigger_reason";
+pub const REGION_MIGRATION_EVENTS_TABLE_TIMEOUT_COLUMN_NAME: &str = "timeout_ms";
+pub const REGION_MIGRATION_EVENTS_TABLE_STATUS_COLUMN_NAME: &str = "status";
+pub const REGION_MIGRATION_EVENTS_TABLE_TIMESTAMP_COLUMN_NAME: &str = "timestamp";
+
+/// RegionMigrationEvent is the event of region migration.
+#[derive(Debug)]
+pub struct RegionMigrationEvent {
+    pub(crate) task: RegionMigrationProcedureTask,
+    pub(crate) procedure_id: ProcedureId,
+    pub(crate) status: RegionMigrationStatus,
+    pub(crate) timestamp: i64,
```

Comment:
Better use our `Timestamp` type in common-time. It has time unit, more readable than a plain `i64` here.